### PR TITLE
Replacing some .formats with F strings

### PR DESCRIPTION
**Pull request by Shimwell**
With the minimal python version soon to be set to python 3.6 in PR #1860 F strings are universally supported. As mentioned in that PR  #1860 one of the benefits is that f strings will now be supported by all supported python version.

This PR replaces the easier to convert .format statements with f string statements.

This PR should only be considered for merging after PR #1860 has been merged

### CODE REVIEW

File: openmc/lattice.py [hunk 1]
```diff
@@ -865,7 +865,7 @@ def create_xml_subelement(self, xml_element, memo=None):
         # Export the Lattice outer Universe (if specified)
         if self._outer is not None:
             outer = ET.SubElement(lattice_subelement, "outer")
-            outer.text = '{0}'.format(self._outer._id)
+            outer.text = f'{self._outer._id}'
```
Comment (paulromano) [inline openmc/lattice.py:868]: This probably ought to just be
```suggestion
            outer.text = str(self._outer._id)
```

File: openmc/lattice.py [hunk 2]
```diff
@@ -1432,7 +1432,7 @@ def create_xml_subelement(self, xml_element, memo=None):
...
```
Comment (paulromano) [inline openmc/lattice.py:1435]:
```suggestion
            outer.text = str(self._outer._id)
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for going through all these!

**Review by paulromano — APPROVED** [review]

**Comment by Shimwell** [comment]
Sorry for accidentality requesting reviews, I've corrected it now.

**Pull request merged**